### PR TITLE
serde 1.0

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by debris** [comment]
nightly compiler is broken, stable and beta are working fine

File: macros/src/auto_args.rs [hunk 1]
```diff
@@ -254,7 +255,7 @@ macro_rules! build_rpc_trait {
 /// A wrapper type without an implementation of `Deserialize`
 /// which allows a special implementation of `Wrap` for functions
 /// that take a trailing default parameter.
-pub struct Trailing<T: Default + Deserialize>(pub T);
+pub struct Trailing<T: DeserializeOwned>(pub T);
```
Comment (tomusdrw) [inline macros/src/auto_args.rs]: `Default` bound has been lost here. I would either remove all the bounds or add `Default` back.

File: macros/src/auto_args.rs [hunk 2]
```diff
@@ -435,8 +436,8 @@ impl<B, OUT, T> WrapAsync<B> for fn(&B, Trailing<T>) -> BoxFuture<OUT, Error>
 	}
 }
 
-impl<B, M, OUT, T> WrapMeta<B, M> for fn(&B, M, Trailing<T>) -> BoxFuture<OUT, Error>
-	where B: Send + Sync + 'static, OUT: Serialize + 'static, T: Default + Deserialize, M: Metadata,
+impl<'a, B, M, OUT, T> WrapMeta<B, M> for fn(&B, M, Trailing<T>) -> BoxFuture<OUT, Error>
```
Comment (tomusdrw) [inline macros/src/auto_args.rs]: Why is the lifetime required here?

File: macros/src/auto_args.rs [hunk 3]
```diff
@@ -422,8 +423,8 @@ impl<B, OUT, T> Wrap<B> for fn(&B, Trailing<T>) -> Result<OUT, Error>
 	}
 }
 
-impl<B, OUT, T> WrapAsync<B> for fn(&B, Trailing<T>) -> BoxFuture<OUT, Error>
-	where B: Send + Sync + 'static, OUT: Serialize + 'static, T: Default + Deserialize
+impl<'a, B, OUT, T> WrapAsync<B> for fn(&B, Trailing<T>) -> BoxFuture<OUT, Error>
```
Comment (tomusdrw) [inline macros/src/auto_args.rs]: Why is the lifetime required here?

File: macros/src/auto_args.rs [hunk 4]
```diff
@@ -412,8 +413,8 @@ fn parse_trailing_param<T: Default + Deserialize>(params: Params) -> Result<(T,
 }
 
 // special impl for no parameters other than block parameter.
-impl<B, OUT, T> Wrap<B> for fn(&B, Trailing<T>) -> Result<OUT, Error>
-	where B: Send + Sync + 'static, OUT: Serialize + 'static, T: Default + Deserialize
+impl<'a, B, OUT, T> Wrap<B> for fn(&B, Trailing<T>) -> Result<OUT, Error>
```
Comment (tomusdrw) [inline macros/src/auto_args.rs]: Why is the lifetime required here?

File: macros/src/auto_args.rs (hunk 1 above)
Comment (debris) [inline macros/src/auto_args.rs]: fixed. If we are about to remove all bounds, I would prefer to do it in a separate pr ;)

File: macros/src/auto_args.rs (hunk 2 above)
Comment (debris) [inline macros/src/auto_args.rs]: they are redundant

**Comment by tomusdrw** [comment]
Closes #123
